src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/FilterDelegationBase.cs

```diff
@@ -187,6 +183,21 @@ protected bool IsValidDelegatableFilterPredicateNode(TexlNode dsNode, TexlBindin
 
             return true;
         }
+
+        private bool IsNodeBooleanOptionSetorBooleanFieldorView(TexlNode dsNode, TexlBinding binding)
+        {
+            // Only boolean option set, boolean fields and views are allowed to delegate
+            var nodeDType = binding.GetType(dsNode);
+            if (binding.IsValidBooleanDelegableNode(dsNode)
+                || (nodeDType == DType.ViewValue))
+            {
+                return true;
+            }
+            else
+            {
+                return false;
+            }
```

Comment:
This could be just

return (binding.IsValidBooleanDelegableNode(dsNode)
                || (nodeDType == DType.ViewValue));